src/internal/Observable.ts

```diff
@@ -251,12 +251,64 @@ export class Observable<T> implements Subscribable<T> {
   }
 
   /**
-   * @method forEach
+   * Used as a NON-CANCELLABLE means of subscribing to an observable, for use with
+   * APIs that expect promises, like `async/await`. You cannot unsubscribe from this.
+   *
+   * **WARNING**: Only use this with observables you *know* will complete. If the source
+   * observable does not complete, you will end up with a promise that is hung up, and
+   * potentially all of the state of an async function hanging out in memory. To avoid
+   * this situation, look into adding something like {@link timeout}, {@link take},
+   * {@link takeWhile}, or {@link takeUntil} amongst others.
+   *
+   * ### Example:
+   *
+   * ```ts
+   * import { interval } from 'rxjs';
+   * import { take } from 'rxjs/operators';
+   *
+   * const source$ = interval(1000).pipe(take(4));
+   *
+   * async function getTotal() {
+   *    let total = 0;
+   *
+   *    await source$.forEach(value => {
+   *      total += value;
+   *      console.log('observable -> ', value);
+   *    });
+   *
+   *    return total;
+   * }
+   *
+   * getTotal().then(
+   *    total => console.log('Total:', total)
+   * )
+   *
+   * // Expected:
+   * // "observable -> 0"
+   * // "observable -> 1"
+   * // "observable -> 2"
+   * // "observable -> 3"
+   * // "Total: 6"
+   * ```
+   * @param next a handler for each value emitted by the observable
+   * @return a promise that either resolves on observable completion or
+   *  rejects with the handled error
+   */
+  forEach(next: (value: T) => void): Promise<void>;
+
+  /**
    * @param next a handler for each value emitted by the observable
-   * @param [promiseCtor] a constructor function used to instantiate the Promise
+   * @param promiseCtor a constructor function used to instantiate the Promise
    * @return a promise that either resolves on observable completion or
    *  rejects with the handled error
+   * @deprecated remove in v8. Passing a Promise constructor will no longer be available
+   * in upcoming versions of RxJS. This is because it adds weight to the library, for very
+   * little benefit. If you need this functionality, it is recommended that you either
+   * polyfill Promise, or you create an adapter to convert the returned native promise
+   * to whatever promise implementation you wanted.
    */
+  forEach(next: (value: T) => void, promiseCtor: PromiseConstructorLike): Promise<void>;
```

Comment:
Why is the `promiseCtor` no longer optional? Adding this signature means that the implementation signature won't be seen, so the fact that the parameter is option in the implementation signature would be meaningless.

Reply:
Never mind. I've misread things.